src/mots/config.py

```diff
@@ -0,0 +1,56 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Configuration classes used to initialize and manage mots in a repo."""
+import logging
+
+from datetime import datetime
+from pathlib import Path
+from ruamel.yaml import YAML
+
+logger = logging.getLogger(__name__)
+
+yaml = YAML()
+yaml.indent(mapping=2, sequence=4, offset=2)
+
+DEFAULT_CONFIG_FILEPATH = "./mots.yaml"
+
+
+class FileConfig:
+    """Loader and writer for filesystem based configuration."""
+
+    def __init__(self, path: Path = DEFAULT_CONFIG_FILEPATH):
+        """Initialize the configuration with provided config file path."""
+        if not path.exists() and path.is_file():
+            raise ValueError(f"{path} does not exist or is not a file.")
+        self.path = path
+        self.repo_path = path.parent
+        self.config = None
```

Comment:
It might be worth following the "internal members behind `_` prefixes" convention, to tighten API surfaces in advance.